lib/abbreviato/truncated_sax_document.rb

```diff
@@ -0,0 +1,177 @@
+require 'nokogiri'
+require 'htmlentities'
+
+class TruncatedSaxDocument < Nokogiri::XML::SAX::Document
+  IGNORABLE_TAGS = %w[html head body].freeze
+
+  # These don't have to be closed (which also impacts ongoing length calculations)
+  # http://www.456bereastreet.com/archive/201005/void_empty_elements_and_self-closing_start_tags_in_html/
+  VOID_TAGS = %w[area base br col command hr img input keygen link meta param source].freeze
```

Comment:
we should compare this with what zendesk comment markup has in terms of void tags...